Faction color in faction configuration popup title

tbeddy opened the pull request:
Closes #48
Alternatively, "with-let" could be changed to a simple "let" and the color information could instead be defined in its bindings.

tbeddy committed:
Removed now redundant button

tbeddy committed:
Added a few ways to implement color (some commented out)

tbeddy committed:
Merge branch 'master' into faction-config-color

Update branch from master.

tbeddy committed:
Merge branch 'faction-config-color' of https://github.com/tbeddy/zetawar into faction-config-color

Merge recent commits from master.

tbeddy committed:
Remove commented out experiments:

tbeddy committed:
Merge branch 'master' into faction-config-color

djwhitt commented on src/cljs/zetawar/data.cljs:
```diff
@@ -18,7 +18,6 @@
 
     ;; Faction configuration
     :configure-faction-tip"Configure faction"
-    :configure-faction-title "Configure faction"
```
I'd keep this and rename it to :configuration-faction-title-prefix. Also, I prefer "Configure faction" to "Configure Faction". I'm not a big fan of title case.

djwhitt commented on src/cljs/zetawar/data.cljs:
```diff
@@ -18,7 +18,6 @@
 
     ;; Faction configuration
     :configure-faction-tip"Configure faction"
```
Not your doing afaik, but we should add a space between the keyword and the string.

djwhitt commented on src/cljs/zetawar/views.cljs:
```diff
@@ -384,7 +384,12 @@
      [:> js/ReactBootstrap.Modal.Header {:close-button true}
       [:> js/ReactBootstrap.Modal.Title
-       (translate :configure-faction-title)]]
+       (str "Configure Faction: "
```
We should still use translate for the title. We should also add translations for the colors (:blue-name, :red-name, etc.) and use them to lookup the color names.

tbeddy replied:
I don't oppose using translate at all, but could you explain why it should be used? Consistency?

djwhitt replied:
I'm not totally consistent yet, but I'd like to use it for all user visible strings. The reason is two fold. First, it gets the strings out of the UI. I think that's generally cleaner (text changes are separated from UI structure changes). Second, it lets us support multiple languages is the future.

djwhitt commented on src/cljs/zetawar/views.cljs:
```diff
@@ -384,7 +384,12 @@
       [:> js/ReactBootstrap.Modal.Title
-       (translate :configure-faction-title)]]
+       (str "Configure Faction: "
+            (when @faction
```
I think I would turn this entire 'when' into a 'deftrack' named 'faction-color-name' in the 'subs' ns (accept conn + faction as args).

tbeddy replied:
What's the reasoning behind having both conn and faction as arguments? Isn't only one necessary to get the desired result?
For example, with faction:
```
(deftrack faction-color-name [faction]
  (when @faction
    (-> @faction ...
```

Or with conn:
```
(deftrack faction-color-name [conn]
  (when-let [faction @(faction-to-configure conn)]
    (-> faction ...
```

Every other deftrack in subs uses conn, so the second would probably be more appropriate.

I might just be missing something obvious.

djwhitt replied:
Nope, not missing anything, you're totally right. I'd go with the first option so that it's useful in contexts other faction configuration.

tbeddy commented:
I'll get back to this in a few days.

tbeddy committed:
First round of edits

tbeddy committed:
Merge branch 'faction-config-color' of https://github.com/tbeddy/zetawar into faction-config-color

Merges upstream changes

djwhitt commented on src/cljs/zetawar/subs.cljs:
```diff
@@ -540,6 +540,23 @@
+    (-> faction
+        :faction/color
+        name
+        (#(apply str [% "-name"]))
```
Can't this just be (str "-name")?

tbeddy replied:
Yeah, it can. Fixed.

tbeddy committed:
More edits:
-Uses faction-color-name deftrack with "faction" as argument
-Moves faction-color-name from the "Faction configuration" section to the more general "Factions" section

tbeddy commented:
Thoughts on my changes?

djwhitt commented on src/cljs/zetawar/subs.cljs:
```diff
@@ -220,6 +220,14 @@
+    (-> @faction
+        :faction/color
+        name
+        (#(str % "-name"))
```
I think you can eliminate the anonymous function altogether can't you? In other words, turn '(#(str % "-name"))" into '(str "-name")'.

tbeddy replied:
Yeah, I misunderstood ->. Fixed.

djwhitt commented:
Sorry, been busy getting up to speed at a new job. The changes look good. I just added one minor comment. I'll merge after that.

tbeddy committed:
Removes unnecessary anonymous function

tbeddy commented:
No problem, good luck with the job.
